.github/workflows/test_and_deploy.yml

```
@@ -58,47 +57,61 @@ jobs:
           retention-days: 7
 
   test_pyver:
-    runs-on: ubuntu-latest
     needs: build
     strategy:
       matrix:
-        pyver:
-          - 3.7
-          - 3.8
+        pyver: [3.6, 3.7, 3.8]
+        runs-on: [macos-latest, ubuntu-latest, windows-latest]
+    runs-on: ${{ matrix.runs-on }}
     steps:
       - name: Download build artifact
         uses: actions/download-artifact@v2
         with:
           name: built_package
           path: ./artifact_storage
-      - name: Setup Python ${{ matrix.pyver }}
-        uses: actions/setup-python@v2
+      - name: Setup Miniconda
+        uses: conda-incubator/setup-miniconda@v2
         with:
-          python-version: ${{ matrix.pyver }}
-      - name: Setup conda env
+          auto-update-conda: true
+          auto-activate-base: false
+          activate-environment: test_env
+      - name: Install dependencies and build artifact
+        shell: bash -l {0}
         run: |
-          source "$CONDA/etc/profile.d/conda.sh"
-          conda config --set always_yes yes --set changeps1 no
-          conda update -q conda
-          conda create -n ci python=${{ matrix.pyver }} pytest conda-build
-          conda activate ci
-          # Setup a local channel
-          BUILT_PKG=`ls ./artifact_storage | head -1`
-          mkdir -p my_channel/noarch
-          mv ./artifact_storage/$BUILT_PKG my_channel/noarch/
-          cd my_channel
-          conda index .
-          cd ..
-          # Install artifact
-          conda install -q -c ./my_channel -c numba numba-extras
+          conda install -c numba python=${{ matrix.pyver }} numba>=0.45 flake8 pytest
+          # Install built_package
+          BUILT_PKG=$(ls ./artifact_storage | head -1)
+          conda install ./artifact_storage/$BUILT_PKG
           conda list
-      - name: numba system check
-        run: |
-          source "$CONDA/etc/profile.d/conda.sh"
-          conda activate ci
-          numba -s
       - name: Pytest
+        shell: bash -l {0}
+        run: |
+          pytest --pyargs numba_extras.tests
```

Testing for this repo is a bit different. Each submodule will be self-contained.

```suggestion
          pytest --pyargs numba_extras
```